jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/AdvanceRuntimeDataServiceImplTest.java

```diff
@@ -682,4 +683,165 @@ public void testParameterWithDash() {
+    }
+
+    @Test
+    public void testQueryProcessByVariablesSortedWithHistory(){
+        List<QueryParam> variables = list(equalsTo("var_b", "1"));
+        List<QueryParam> attributes = list(history());
+        assertQueryProcessByVariablesSorted(attributes, variables);
+    }
+
+    @Test
+    public void testQueryUserTaskByVariablesSorted() {
+        List<QueryParam> variables = list(equalsTo("task_in_a1", "a0"));
+        List<QueryParam> attributes = list(equalsTo(PROCESS_ATTR_DEPLOYMENT_ID, "org.jbpm.test:test-module:1.0.0"));
+        assertQueryUserTaskByVariablesSorted(attributes, variables);
+    }
+
+    @Test
+    public void testQueryUserTaskByVariablesSortedWithHistory() {
+        List<QueryParam> variables = list(equalsTo("task_in_a1", "a0"));
+        List<QueryParam> attributes = list(history());
+        assertQueryUserTaskByVariablesSorted(attributes, variables);
+    }
+
+    @Test
+    public void testQueryProcessByVariablesAndTaskSorted() {
+        List<QueryParam> variables = list(equalsTo("var_b", "1"));
+        List<QueryParam> attributes = list(in(PROCESS_ATTR_DEFINITION_ID, "test.test_A", "test.test_B"));
+        assertQueryProcessByVariablesAndTaskSorted(attributes, variables);
+    }
+
+    @Test
+    public void testQueryProcessByVariablesAndTaskSortedWithHistory() {
+        List<QueryParam> variables = list(equalsTo("var_b", "1"));
+        List<QueryParam> attributes = list(history());
+        assertQueryProcessByVariablesAndTaskSorted(attributes, variables);
+    }
+
+    private void assertQueryProcessByVariablesSorted(List<QueryParam> attributes, List<QueryParam> variables) {
+        List<ProcessInstanceWithVarsDesc> actualDataList = advanceVariableDataService.queryProcessByVariables(attributes, variables, new QueryContext());
+        List<Long> expectedProcessInstanceIdsList = actualDataList
+                .stream()
+                .map(ProcessInstanceWithVarsDesc::getId)
+                .collect(Collectors.toList());
+        List<Date> expectedStartDatesList = actualDataList
+                .stream()
+                .map(ProcessInstanceWithVarsDesc::getDataTimeStamp)
+                .collect(Collectors.toList());
+        assertThat(expectedProcessInstanceIdsList.size(), is(6));
+        assertThat(expectedStartDatesList.size(), is(6));
+
+        // Order by processInstanceId ascending
+        queryContext.setOrderBy("processInstanceId");
+        queryContext.setAscending(true);
+        actualDataList = advanceVariableDataService.queryProcessByVariables(attributes, variables, queryContext);
+        assertSortedListByColumn(true, expectedProcessInstanceIdsList, actualDataList.stream().map(ProcessInstanceWithVarsDesc::getId).collect(Collectors.toList()));
+
+        // Order by processInstanceId descending
+        queryContext.setAscending(false);
+        actualDataList = advanceVariableDataService.queryProcessByVariables(attributes, variables, queryContext);
```

Comment:
to make this clear to read (at least for me, this is very opinnable), maybe rather than reusing actualDataLIst, you can just write  advanceVariableDataService.queryProcessByVariables(attributes, variables, queryContext).stream()......

Reply:
agree. Done